### PR TITLE
enhance unit test  for non canonical encoding length

### PR DESCRIPTION
**Pull request by Xiaohanjlll**

### CODE REVIEW

File: unit_test/test_spdm_crypt/test_spdm_crypt.c [hunk 1]
```diff
@@ -35,6 +35,112 @@ uint8_t m_libspdm_subject_alt_name_buffer3[] = {
 uint8_t m_libspdm_dmtf_oid[] = { 0x2B, 0x06, 0x01, 0x4,  0x01,
                                  0x83, 0x1C, 0x82, 0x12, 0x01 };
 
+/* rsa2048/end_requester.cert.der
+ * https://lapo.it/asn1js/#MIIEDzCCAnegAwIBAgIBAzANBgkqhkiG9w0BAQsFADAtMSswKQYDVQQDDCJETVRGIGxpYnNwZG... */
+uint8_t m_cert_buffer[] = {
```
Comment (jyao1) [inline unit_test/test_spdm_crypt/test_spdm_crypt.c:40]: please dont hardcode buffer. Please read file.

File: unit_test/test_spdm_crypt/test_spdm_crypt.c [hunk 2]
```diff
@@ -401,6 +507,36 @@ void libspdm_test_crypt_spdm_x509_certificate_check(void **state)
...
+                                                true, true);
+        assert_true(status);
+
+        /* The definite form of length encoding shall be used, encoded in the minimum number of octets
+         * Use non canonical encoding format , length encoding is : 0x81 0x03*/
+        data = 0x81;
+        insert_data(buffer, buffer_size, &data, sizeof(data), 9);
+        buffer_size += sizeof(data);
+        buffer[3] += sizeof(data);
+        buffer[7] += sizeof(data);
```
Comment (jyao1) [inline unit_test/test_spdm_crypt/test_spdm_crypt.c:532]: what is buffer[3] and buffer[7]?

File: unit_test/test_spdm_crypt/test_spdm_crypt.c (hunk 1 above)
Comment (Xiaohanjlll) [reply unit_test/test_spdm_crypt/test_spdm_crypt.c]: Thank , I have modified it

File: unit_test/test_spdm_crypt/test_spdm_crypt.c (hunk 2 above)
Comment (Xiaohanjlll) [reply unit_test/test_spdm_crypt/test_spdm_crypt.c]: this is of Certificate SEQUENCE  length and tbsCertificate SEQUENCE length

**Pull request closed without merging**